src/hotspot/share/oops/resolvedFieldEntry.cpp

```diff
@@ -23,8 +23,17 @@
  */
 
 #include "cds/archiveBuilder.hpp"
+#include "cppstdlib/type_traits.hpp"
 #include "oops/resolvedFieldEntry.hpp"
 
+STATIC_ASSERT(std::is_trivially_copyable_v<ResolvedFieldEntry> == true);
+
+// Detect inadvertently introduced trailing padding.
```

Comment:
Another way to detect padding (trailing or internal) could be to compare sizeof(ResolvedFieldEntry) against the sum up the size of all the elements. Something like:
`sizeofResolvedFieldEntry) == (sizeof(InstanceKlass*) + sizeof(int) + sizeof(u2) + ... + sizeof(_padding))`
It looks cumbersome but I think it is easy enough to update if a new field is added because the static assert will fail immediately.